query/src/main/java/org/infinispan/query/concurrent/InfinispanIndexingExecutorProvider.java

```diff
@@ -0,0 +1,35 @@
+package org.infinispan.query.concurrent;
+
+import java.util.concurrent.ScheduledExecutorService;
+
+import org.hibernate.search.backend.lucene.work.spi.LuceneWorkExecutorProvider;
+import org.hibernate.search.engine.common.execution.spi.DelegatingSimpleScheduledExecutor;
+import org.hibernate.search.engine.common.execution.spi.SimpleScheduledExecutor;
+import org.infinispan.factories.AbstractComponentFactory;
+import org.infinispan.factories.AutoInstantiableFactory;
+import org.infinispan.factories.KnownComponentNames;
+import org.infinispan.factories.annotations.ComponentName;
+import org.infinispan.factories.annotations.DefaultFactoryFor;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+
+@Scope(Scopes.GLOBAL)
+@DefaultFactoryFor(classes = LuceneWorkExecutorProvider.class)
+public class InfinispanIndexingExecutorProvider extends AbstractComponentFactory implements AutoInstantiableFactory, LuceneWorkExecutorProvider {
+
+   // injecting the global *blocking* scheduled executor
+   @ComponentName(KnownComponentNames.TIMEOUT_SCHEDULE_EXECUTOR)
+   @Inject
+   ScheduledExecutorService scheduledExecutorService;
+
+   @Override
+   public Object construct(String name) {
+      return new InfinispanIndexingExecutorProvider();
+   }
+
+   @Override
+   public SimpleScheduledExecutor writeExecutor(Context context) {
+      return new DelegatingSimpleScheduledExecutor(scheduledExecutorService);
```

Comment:
I assume this is going to change later? We do not want to pass along the scheduled executor like this. We would only want to schedule a task that in turn submits the task to the blocking executor. This way scheduled executor service is not blocked.

Reply:
> We would only want to schedule a task that in turn submits the task to the blocking executor.

I changed a bit the implementation implementing our SimpleScheduledExecutor (that is `InfinispanIndexingExecutorProvider.InfinispanScheduledExecutor`).
I also injected the blocking executor and delegated to it all non-scheduled tasks.
For the scheduled tasks we still delegate to the `TIMEOUT_SCHEDULE_EXECUTOR`.

If we need to change this behavior, I think that is better to do now, before Hibernate Search 6.2.Alpha2 is released.

So, how can we  `schedule a task that in turn submits the task to the blocking executor`?
Thank you Will